grpc-server-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/server/health/GrpcServerActuatorHealthIndicator.java

```
@@ -0,0 +1,100 @@
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package net.devh.boot.grpc.server.health;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.springframework.boot.actuate.health.AbstractReactiveHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.Status;
+
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.MoreExecutors;
+
+import io.grpc.Channel;
+import io.grpc.Deadline;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.health.v1.HealthCheckRequest;
+import io.grpc.health.v1.HealthCheckResponse;
+import io.grpc.health.v1.HealthGrpc;
+import reactor.core.publisher.Mono;
+
+/**
+ * This class will expose the status of the server into the actuator when it has the "show-components" configuration
+ * set. To retrieve the status the health probe actually hits the server and maps the status
+ */
+public class GrpcServerActuatorHealthIndicator extends AbstractReactiveHealthIndicator {
+
+    private final static String STATUS = "status";
+    private final static String DOWN_STATUS = "DOWN";
+    private HealthGrpc.HealthFutureStub healthCheckClient;
+    private long deadlineMs = 500;
+
+    public GrpcServerActuatorHealthIndicator(String host, int port, long deadline) {
+        String internalHost = host.equals("*") ? "0.0.0.0" : host;
+        Channel channel = ManagedChannelBuilder
+                .forAddress(internalHost, port)
+                .disableRetry()
+                .build();
+        healthCheckClient = HealthGrpc.newFutureStub(channel);
+        if (deadline > 0) {
+            this.deadlineMs = deadline;
+        }
```

IMO this should fail, if the deadline is invalid.